kolena/_utils/batched_load.py

```diff
@@ -79,6 +82,18 @@ def upload_data_frame_chunk(df_chunk: pd.DataFrame, load_uuid: str) -> None:
 DFType = TypeVar("DFType", bound=LoadableDataFrame)
 
 
+def upload_smart_chunk_data_frame(df: pd.DataFrame, uuid: str, rows: int = 1000) -> None:
+    batch_size = (BATCH_LIMIT // get_preflight_export_size(df, rows)) * rows
+    upload_data_frame(df, batch_size, uuid)
+
+
+def get_preflight_export_size(df: pd.DataFrame, rows: int) -> int:
+    df_subset = df[:rows]
+    with tempfile.NamedTemporaryFile() as temp:
+        df_subset.to_parquet(temp.name)
+        return os.stat(temp.name).st_size
```

Comment:
could we add some defensive programming to raise error when the size is 0?

i understand that it should never be 0 in practice, but from the programming perspective it's possible to be 0

Reply:
will update

Reply:
done